File: src/main/java/cn/mobiledaily/web/managedbean/ExhibitionBean.java
package cn.mobiledaily.web.managedbean;

import cn.mobiledaily.domain.Exhibition;
import cn.mobiledaily.service.ExhibitionService;
import cn.mobiledaily.web.common.SpringContext;

import javax.faces.application.FacesMessage;
import javax.faces.bean.ManagedBean;
import javax.faces.bean.ManagedProperty;
import javax.faces.bean.ViewScoped;
import javax.faces.context.FacesContext;
import javax.faces.event.ActionEvent;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.Serializable;
import java.util.List;

@ManagedBean(name = "exhibition")
@ViewScoped
public class ExhibitionBean implements Serializable {
    @ManagedProperty("#{exhibitionService}")
    transient private ExhibitionService exhibitionService;
    private Exhibition item;
    private boolean edit;

    private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
        in.defaultReadObject();
        exhibitionService = SpringContext.getExhibitionService();
    }

    public void setExhibitionService(ExhibitionService exhibitionService) {
        this.exhibitionService = exhibitionService;
    }

    public List<Exhibition> getItems() {
        return exhibitionService.findAll();
    }

    public Exhibition getItem() {
        return item;
    }

    public void save(ActionEvent actionEvent) {
        try {
            exhibitionService.save(item);
            item = null;
        } catch (Exception e) {
            FacesContext.getCurrentInstance().addMessage(null, new FacesMessage(FacesMessage.SEVERITY_FATAL, "Internal Error", e.getMessage()));
        }
    }

    public void edit(String id) {
        item = exhibitionService.findById(id);
        edit = true;
    }

    public void create() {
        if (edit || item == null) {
            item = createItem();
        }
    }

    public void remove(ActionEvent actionEvent) {
        exhibitionService.delete(item);
    }

    private Exhibition createItem() {
        Exhibition exhibition = new Exhibition();
        edit = false;
        return exhibition;
    }
}

File: src/main/scala/cn/mobiledaily/web/mbean/ExhibitionBean.scala
package cn.mobiledaily.web.mbean

import javax.faces.bean.{ViewScoped, ManagedBean}
import cn.mobiledaily.web.common.SpringContext
import cn.mobiledaily.domain.Exhibition
import javax.faces.event.ActionEvent
import javax.faces.context.FacesContext
import javax.faces.application.FacesMessage

@ManagedBean(name = "exhibition")
@ViewScoped
class ExhibitionBean extends Serializable {
    val exhibitionService = SpringContext.exhibitionService
    var item: Exhibition = null
    private var isEdit: Boolean = false

    def getItems = exhibitionService.findAll

    def getItem = item

    def save(event: ActionEvent) {
        try {
            exhibitionService.save(item)
            item = null
        } catch {
            case e: Exception => FacesContext.getCurrentInstance.addMessage(
                null, new FacesMessage(FacesMessage.SEVERITY_FATAL, "Internal Error", e.getMessage)
            )
        }
    }

    def edit(id: String) {
        item = exhibitionService.findById(id)
        isEdit = true
    }

    def create() {
        if (isEdit || item == null) {
            item = createItem
        }
    }

    def remove(event: ActionEvent) {
        exhibitionService.delete(item)
    }

    private def createItem = {
        isEdit = false
        new Exhibition()
    }
}
